src/main/java/ca/ciccc/BankAccount.java

```diff
@@ -75,72 +76,110 @@ public void withdraw(double amount) {
      * afford all of the fees or has no money left, the balance is left as 0.0 and method returns
      * {@code false}.
-     *
+     * <p>
      * For example, given the following ca.ciccc.BankAccount object.
-     *
+     * <p>
      * ca.ciccc.BankAccount savings = new ca.ciccc.BankAccount("Jimmy");
      * savings.deposit(10.00);
      * savings.deposit(50.00);
      * savings.deposit(10.00);
      * savings.deposit(70.00);
-     *
+     * <p>
      * savings.transactionFee(5.00);
-     *
+     * <p>
      * The account would be deducted $5 + $10 + $15 + $20 for the four transactions, leaving a final
      * balance of $90.00. The method would return true.
-     *
+     * <p>
      * savings.transactionFee(10.00);
-     *
+     * <p>
      * Then the account would be deducted $10 + $20 + $30 + $40 for the four transactions, leaving
      * a final balance of $0.00. The method would return false.
      *
      * @param fee
      * @return true if there's enough balance, otherwise false
      */
     public boolean transactionFee(double fee) {
-        // TODO 2: Your code goes here.
+        double totalfee = 0;
+
+        for (int i = 1; i < transactionCount; i++) {
+            totalfee = totalfee + (i * fee);
+
+            balance = balance - totalfee;
+            if (balance < 0) {
+                return false;
+            }
+
+        }
+        return true;
 
-        return false;
     }
 
+
     /**
      * The transfer method moves money from `this` bank account to another account. The method
      * accepts two parameters: the amount of money to transfer and the other ca.ciccc.BankAccount to accept
      * the money.
      * There is a $5.00 fee for transferring money, so this much must be deducted from the current
      * account's balance before any transfer.
-     *
+     * <p>
      * If `this` account object does not have enough money to make full transfer, then transfer
      * whatever money is left after the $5.00 fee is deducted. If this account has under $5.00 or amount is 0 or less,
      * no transfer should occur and neither account's state should be modified.
-     *
+     * <p>
      * If any amount of money is transferred, return {@code true}. Otherwise {@code false}.
      *
      * @param amount
      * @param other
      * @return true if transferred any amount of money, otherwise false.
      */
-    public boolean transfer(double amount, BankAccount other) {
-        // TODO 3: Your code goes here.
+    public String transfer(double amount, BankAccount other) {
+        double fee = 5.00;
+        double total = 0;
+        double BankAccount = this.getBalance();
+        for (int i = 1; i < transactionCount; i++) {
+            total = amount + (i * fee);
+            if (BankAccount - total >= 0) {
+                this.withdraw(fee);
+                this.withdraw(amount);
+                other.withdraw(amount);
+                other.deposit(amount);
+                return String.valueOf(true);
```

Comment:
[FYI] You can just return `true`.

```suggestion
                return true;
```